[BUG] Correct 'StarlightCurves' data set identifier string, to 'StarLightCurves'

NeuralNut opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look
at our contribution guide: https://github.com/sktime/sktime/blob/main/CONTRIBUTING.md
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.

Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->


#### What does this implement/fix? Explain your changes.
<!--
A clear and concise description of what you have implemented.
-->

#### Does your contribution introduce a new dependency? If yes, which one?

<!--
If your contribution does add a new hard dependency, we may suggest to initially develop your contribution in a separate companion package in https://github.com/sktime/ to keep external dependencies of the core sktime package to a minimum.
-->

#### What should a reviewer concentrate their feedback on?

<!-- This section is particularly useful if you have a pull request that is still in development. You can guide the reviews to focus on the parts that are ready for their comments. We suggest using bullets (indicated by * or -) and filled checkboxes [x] here -->

#### Did you add any tests for the change?

<!-- This section is useful if you have added a test in addition to the existing ones. This will ensure that further changes to these files won't introduce the same kind of bug. It is considered good practice to add tests with newly added code to enforce the fact that the code actually works. This will reduce the chance of introducing logical bugs.
-->

#### Any other comments?
<!--
Please be aware that we are a loose team of volunteers so patience is necessary; assistance handling other issues is very welcome. We value all user contributions, no matter how minor they are. If we are slow to review, either the pull request needs some benchmarking, tinkering, convincing, etc. or more likely the reviewers are simply busy. In either case, we ask for your understanding during the review process.
-->

#### PR checklist
<!--
Please go through the checklist below. Please feel free to remove points if they are not applicable.
-->

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/sktime/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.


<!--
Thanks for contributing!
-->

fkiraly reviewed (approved):
Thanks!

Can you quickly explain how you detected the issue?
Do you know why this was not detected by a test, for instance?

NeuralNut commented:
@fkiraly Yes I ran the following just to pre-download all the univariate equal length UEA datasets
```Python
from sktime.datasets.tsc_dataset_names import univariate_equal_length
for dataset in univariate_equal_length:
    _ = load_UCR_UEA_dataset(dataset)
```
Without the corrected StarLightCurves dataset name within the `univariate_equal_length` list, the following error was produced
```Python
---------------------------------------------------------------------------
BadZipFile                                Traceback (most recent call last)
File ~/miniconda3/envs/cplx/lib/python3.8/site-packages/sktime/datasets/_data_io.py:102, in _download_and_extract(url, extract_path)
    101     os.makedirs(extract_path)
--> 102 zipfile.ZipFile(zip_file_name, "r").extractall(extract_path)
    103 shutil.rmtree(dl_dir)

File ~/miniconda3/envs/cplx/lib/python3.8/zipfile.py:1269, in ZipFile.__init__(self, file, mode, compression, allowZip64, compresslevel, strict_timestamps)
   1268 if mode == 'r':
-> 1269     self._RealGetContents()
   1270 elif mode in ('w', 'x'):
   1271     # set the modified flag so central directory gets written
   1272     # even if no files are added to the archive

File ~/miniconda3/envs/cplx/lib/python3.8/zipfile.py:1336, in ZipFile._RealGetContents(self)
   1335 if not endrec:
-> 1336     raise BadZipFile("File is not a zip file")
   1337 if self.debug > 1:

BadZipFile: File is not a zip file

During handling of the above exception, another exception occurred:

BadZipFile                                Traceback (most recent call last)
File ~/miniconda3/envs/cplx/lib/python3.8/site-packages/sktime/datasets/_data_io.py:202, in _load_dataset(name, split, return_X_y, return_type, extract_path)
    201 try:
--> 202     _download_and_extract(
    203         url,
    204         extract_path=extract_path,
    205     )
    206 except zipfile.BadZipFile as e:

File ~/miniconda3/envs/cplx/lib/python3.8/site-packages/sktime/datasets/_data_io.py:109, in _download_and_extract(url, extract_path)
    108     shutil.rmtree(extract_path)
--> 109 raise zipfile.BadZipFile(
    110     "Could not unzip dataset. Please make sure the URL is valid."
    111 )

BadZipFile: Could not unzip dataset. Please make sure the URL is valid.

The above exception was the direct cause of the following exception:

ValueError                                Traceback (most recent call last)
Cell In[40], line 7
      5 for dataset in univariate_equal_length:
      6     print(dataset)
----> 7     _ = load_UCR_UEA_dataset(dataset)

File ~/miniconda3/envs/cplx/lib/python3.8/site-packages/sktime/datasets/_single_problem_loaders.py:119, in load_UCR_UEA_dataset(name, split, return_X_y, return_type, extract_path)
     62 def load_UCR_UEA_dataset(
     63     name, split=None, return_X_y=True, return_type=None, extract_path=None
     64 ):
     65     """Load dataset from UCR UEA time series archive.
     66 
     67     Downloads and extracts dataset if not already downloaded. Data is assumed to be
   (...)
    117     >>> X, y = load_UCR_UEA_dataset(name="ArrowHead")
    118     """
--> 119     return _load_dataset(name, split, return_X_y, return_type, extract_path)

File ~/miniconda3/envs/cplx/lib/python3.8/site-packages/sktime/datasets/_data_io.py:207, in _load_dataset(name, split, return_X_y, return_type, extract_path)
    202             _download_and_extract(
    203                 url,
    204                 extract_path=extract_path,
    205             )
    206         except zipfile.BadZipFile as e:
--> 207             raise ValueError(
    208                 f"Invalid dataset name ={name} is not available on extract path ="
    209                 f"{extract_path}. Nor is it available on "
    210                 f"https://timeseriesclassification.com/.",
    211             ) from e
    213 return _load_provided_dataset(
    214     name, split, return_X_y, return_type, local_module, local_dirname
    215 )

ValueError: Invalid dataset name =StarlightCurves is not available on extract path =None. Nor is it available on https://timeseriesclassification.com/.
```

I guess none of the tests use `univariate_equal_length` ?

fkiraly commented:
hm, maybe not all the data sets are covered for download in the tests, only some.
